tls/s2n_config.h

```diff
@@ -109,3 +109,5 @@ extern int s2n_config_free_session_ticket_keys(struct s2n_config *config);
 extern void s2n_wipe_static_configs(void);
 extern struct s2n_cert_chain_and_key *s2n_config_get_single_default_cert(struct s2n_config *config);
 extern int s2n_config_get_num_default_certs(struct s2n_config *config);
+
+int s2n_config_add_all_curves(struct s2n_config *conf);
```

Comment:
I am curious on what this API was intended for

Reply:
That should have been deleted. The original intent was to add all supported curves to the list of curves in the config object. But this all depends on how we want to expose this functionality to customers, so I removed it.